exporter/signalfxexporter/translation/logdata_to_signalfxv2.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translation
+
+import (
+	"fmt"
+
+	sfxpb "github.com/signalfx/com_signalfx_metrics_protobuf/model"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/splunk"
+)
+
+func LogSliceToSignalFxV2(
+	logger *zap.Logger,
+	logs pdata.LogSlice,
+) ([]*sfxpb.Event, int) {
+	var events []*sfxpb.Event
```

Comment:
1. Preallocate the slice of pointers to reduce allocations?
2. It is also possible to preallocate a slice of sfxpb.Event structs to reduce allocations even further.
(I don't know how much a bottleneck is it in this particular case but usually allocations show as a big chunk in our profiles).

Reply:
I preallocated the slice of pointers but I doubt events will ever be in such high volumes to make it worth writing it in C-style.  We generally don't like to get more than a few events per second at most on average.